flytekit/core/type_engine.py

```diff
@@ -1187,7 +1188,7 @@ def dict_to_literal_map(
         ctx: FlyteContext,
         d: typing.Dict[str, typing.Any],
         type_hints: Optional[typing.Dict[str, type]] = None,
-    ) -> LiteralMap:
+    ) -> Optional[LiteralMap]:
```

Comment:
Why Optional?

Reply:
Revert it. Good catch. I forgot why I made it optional.